controller/array_action/array_mediator_svc.py

```diff
@@ -148,30 +164,65 @@ def _generate_volume_response(self, cli_volume):
             cli_volume.mdisk_grp_name,
             self.array_type)
 
-    def get_volume(self, volume_name, volume_context=None, volume_prefix=""):
-        logger.debug("Get volume : {}".format(volume_name))
-        cli_volume = None
+    def _generate_snapshot_response(self, cli_snapshot, source_volume_name):
+        return Snapshot(int(cli_snapshot.capacity),
+                        cli_snapshot.vdisk_UID,
+                        cli_snapshot.name,
+                        self.endpoint,
+                        source_volume_name,
+                        is_ready=True,
+                        array_type=self.array_type)
+
+    def _get_client_list_method(self, cli_object_type):
+        svcinfo = self.client.svcinfo
+        if cli_object_type == CLI_OBJECT_TYPE_VOLUME:
+            return lambda obj_id: svcinfo.lsvdisk(bytes=True, object_id=obj_id)
+        if cli_object_type == CLI_OBJECT_TYPE_FCMAP:
+            return lambda obj_id: svcinfo.lsfcmap(object_id=obj_id)
+        raise ValueError("No list method for cli object type: {}".format(cli_object_type))
+
+    def _get_cli_object(self, object_name_or_id, cli_object_type, not_exist_err=True):
```

Comment:
This method makes the code much less clear and readable.
Make separate methods -  one for volume and one for snapshot

Reply:
I'm sorry to hear that it's not clear enough, but we shouldn't just keep duplicating (17~ lines) functions either.
Would it help if I call a generic list function, without returning it first?
e.g.
```
def list_cli_object(self, cli_object_type):
    if cli_object_type == CLI_OBJECT_TYPE_VOLUME:
        return svcinfo.lsvdisk(...)list_method(object_name_or_id).as_single_element
    ...
```

Reply:
No. list_cli_object is bad method.
Too much generalization makes code much less readable.

Reply:
could we get a 3rd opinion on this? @ArbelNathan?
I think it's readable enough, and otherwise `_get_cli_volume` & `_get_fcmap` would be ~99% same code

Reply:
I think that the method _get_fcmap that already exist is doing the job of explaining..

Reply:
thanks @ArbelNathan 
@igorgonibm would you like to propose another suggestion? or should we resolve this?

Reply:
I disagree.

Reply:
I disagree